### ISSUE TITLE
question about tf.stop_gradient in deepwalk test part

### ISSUE THREAD

**Issue by wierdoki** [williamleif/GraphSAGE#36]
isn't it wrong to simply assign model.context_embeds with update_nodes+no_update_nodes, where no_update_nodes = tf.stop_gradient(...), for which you meant to stop the gradient for the already trained nodes? because the old model.context_embeds is still there in the tf graph, used by model.opt_op, right? plz correct me if I'm wrong, thx.

**Comment by RexYing** [comment]
In the inductive setting, only inference is allowed. Of course one can re-train the model by fine-tuning the existing embeddings, but that's often very expensive in large graphs: altering any existing embedding would require updating potentially all embeddings, due to links that exist between them.
